### PR TITLE
Log exceptions in TcpTransport at DEBUG level

### PR DESCRIPTION
**Pull request by DaveCTurner**
When running Elasticsearch on a flaky network, we may see nodes leaving the
cluster with reason `disconnected`. It may be useful to the cluster
administrator to see the full exception that caused the disconnection, but this
is only available with `TRACE` level logging which commingles the details of
the problem with other messages that are not useful to end users.

This commit promotes logging of exceptions in `TcpTransport` from `TRACE` to
`DEBUG` to separate them from the truly `TRACE`-level messages.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/transport/TcpTransportTests.java [hunk 1]
```diff
@@ -354,4 +366,101 @@ public void testHTTPResponse() throws IOException {
...
+                assertFalse(listener.isDone());
+            }
+
+            appender.assertAllExpectationsMatched();
+
+        } finally {
+            Loggers.removeAppender(LogManager.getLogger(TcpTransport.class), appender);
+            appender.stop();
+            IOUtils.close(tcpTransport);
+            testThreadPool.shutdown();
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/transport/TcpTransportTests.java]: NIT: Maybe use `org.elasticsearch.threadpool.ThreadPool#terminate(java.util.concurrent.ExecutorService, long, java.util.concurrent.TimeUnit)` here so we don't get the leaked thread warning when running this one?

**Review by original-brownbear — APPROVED** [review]
Makes sense IMO => LGTM

File: server/src/test/java/org/elasticsearch/transport/TcpTransportTests.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/transport/TcpTransportTests.java]: TIL. Yes, done in f0e813b23be.

**Review by Tim-Brooks — CHANGES_REQUESTED** [review]
If we are going to implement these tests can we extract the exception handling to a separate class and test in a manner that does not require overriding the transport? See `SecurityHttpExceptionHandler` as a place where we extracted exception handling from the http transport.

**Comment by DaveCTurner** [comment]
Not sure of the value of a whole nother class for this, how about a static method? See a62032c.

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/default-distro

**Pull request merged**